rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -317,33 +317,77 @@ public String toString() {
 
     @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
     @EqualsAndHashCode(callSuper = false, onlyExplicitlyIncluded = true)
+    @RequiredArgsConstructor
+    @With
     @Data
     final class ArrayType implements J, TypeTree, Expression {
-        @With
         @EqualsAndHashCode.Include
         UUID id;
 
-        @With
         Space prefix;
-
-        @With
         Markers markers;
-
-        @With
         TypeTree elementType;
 
-        @With
-        List<JRightPadded<Space>> dimensions;
+        @Nullable
+        List<J.Annotation> annotations;
 
-        @Override
-        public JavaType getType() {
-            return elementType.getType();
-        }
+        JLeftPadded<Space> dimension;
 
-        @SuppressWarnings("unchecked")
-        @Override
-        public ArrayType withType(@Nullable JavaType type) {
-            return type == getType() ? this : withElementType(elementType.withType(type));
+        @Nullable
+        JavaType type;
+
+        @Deprecated
+        @JsonCreator
+        public ArrayType(UUID id,
```

Comment:
I suggest we declare it package private. Possibly we should also add a comment on the `dimensions` parameter (or a `@JsonProperty` annotation), just so that nobody accidentally removes or renames it.